File: tagalong/005-mulreturn/mulreturn.zig
const std = @import("std");

const result = struct {
    down: i64,
    up: i64,
};

pub fn collatz(a:i64) result {
    return result{
        .down = @divFloor(a, 2),
        .up   = a*3+1,
    };
}

pub fn main() void {
    const v:i64 = 60;
    var r = collatz(v);
    r.up = 2;
    std.debug.print("empezando en {} hay que saber subir {}, y bajar {}\n", .{v, r.up, r.down});
}

File: tagalong/005-mulreturn/mulreturn.py
def collatz(a:int) -> tuple[int, int]:
    down = a//2
    up = 3*a+1
    return down, up

v = 60
down, up = collatz(v)
print(f"empezando en {v} hay que saber subir {up}, y bajar {down}")